integration_tests/test_upload.py

```diff
@@ -48,3 +51,25 @@ def test_upload(self):
         self.assertIn('Upload failed', output)
         self.assertIn('No valid credentials found. Have you run "snapcraft '
                       'login"?', output)
+
+    def test_upload_with_login(self):
+        self.useFixture(fixture_setup.StagingStore())
+
+        project_dir = 'basic'
+        self.addCleanup(self.logout)
+        self.login()
+
+        project_dir = self.copy_project_to_tmp(project_dir)
+        yaml_file = os.path.join(project_dir, 'snapcraft.yaml')
+        for line in fileinput.input(yaml_file, inplace=True):
+            if 'version: ' in line:
+                # Change to a random version.
+                # The maximum size is 32 chars.
+                print('version: ' + str(uuid.uuid4().int)[:32])
```

Comment:
Not as far as I can see. It assigns a sequential version starting from 1, ignoring what we pass in this string. I'd say we have some tens of thousands of executions before worrying. And I hope that before we hit that many executions in staging we have a way to fake or deploy the server locally, so we will have to run fewer slow tests hitting staging.